.github/workflows/publish.yml

```diff
@@ -1,59 +1,123 @@
+          path: "${{ steps.temp-dir.outputs.path }}"
+          sha256: ${{ needs.provenance.outputs.provenance-download-sha256 }}
+
+      - name: Unpack the zipped artifact
+        run: |
+          set -euo pipefail
+          cd "${{ steps.temp-dir.outputs.path }}"
+          tar -xzvf "${{ needs.provenance.outputs.package-name }}" -C $GITHUB_WORKSPACE --strip-components=1
+          cd "$GITHUB_WORKSPACE"
+          pnpm run install-deps
+          pnpm nx-test-skip-cache
+
+      - name: Create Release Pull Request or Publish to npm
+        id: changesets
+        uses: changesets/action@aba318e9165b45b7948c60273e0b72fce0a64eb9 # v1.4.7
+        with:
+          setupGitUser: false
+          version: pnpm ci:version
+          title: "ci: Update the version packages"
+          publish: pnpm release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Generate SBOM
+        uses: anchore/sbom-action@95b086ac308035dc0850b3853be5b7ab108236a8
+        with:
+          artifact-name: sbom-${{ github.event.repository.name }}-${{ inputs.version_tag }}.spdx.json
+          output-file: /${{ steps.temp-dir.outputs.path }}/sbom-${{ github.event.repository.name }}-${{ inputs.version_tag }}.spdx.json
+          upload-artifact: false
+          upload-release-assets: false
+
+      - name: Download Artifacts
+        uses: actions/download-artifact@fb598a63ae348fa914e94cd0ff38f362e927b741
+
+      - name: Upload attestations SLSA
+        if: steps.changesets.outputs.id != ''
+        uses: actions/attest-build-provenance@5e9cb68e95676991667494a6a4e59b8a2f13e1d0
         with:
-          node-version: 20.11.1
+          subject-path: ${{ needs.provenance.outputs.provenance-download-name }}
+          subject-name: ${{ github.event.repository.name }}-${{ inputs.version_tag }}
 
-      - name: Publish to NPM
-        id: publish
-        uses: slsa-framework/slsa-github-generator/actions/nodejs/publish@41733f74c025cc6d156547121989dd50fbc92364 # v2.0.0.pre.rc.0
+      - name: Upload attestations SBOM
+        if: steps.changesets.outputs.id != ''
+        uses: actions/attest-build-provenance@5e9cb68e95676991667494a6a4e59b8a2f13e1d0
         with:
-          access: public
-          node-auth-token: ${{ secrets.NPM_TOKEN }}
-          package-name: ${{ needs.provenance.outputs.package-name }}
-          package-download-name: ${{ needs.provenance.outputs.package-download-name }}
-          package-download-sha256: ${{ needs.provenance.outputs.package-download-sha256 }}
-          provenance-name: ${{ needs.provenance.outputs.provenance-name }}
-          provenance-download-name: ${{ needs.provenance.outputs.provenance-download-name }}
-          provenance-download-sha256: ${{ needs.provenance.outputs.provenance-download-sha256 }}
+          subject-path: /${{ steps.temp-dir.outputs.path }}/sbom-${{ github.event.repository.name }}-${{ inputs.version_tag }}.spdx.json
```

Comment:
Is `git push --follow-tags` missing here? can we add it to changeset actions somehow to push the tags automatically after publish or do we need an extra step here.